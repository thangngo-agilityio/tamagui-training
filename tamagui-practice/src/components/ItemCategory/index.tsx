import { Flex, Heading, Radio } from '@chakra-ui/react';
import { memo, ReactElement } from 'react';
import isEqual from 'react-fast-compare';

type TItemCategory = {
  title: string;
  value?: string;
  filter?: string;
  icon?: ReactElement;
  iconActive?: ReactElement;
};

const ItemCategory = ({
  title,
  filter,
  value,
  icon,
  iconActive,
}: TItemCategory) => {
  const isActive = value === filter;

  return (
    <Radio m={0} value={value}>
      <Flex
        w="56px"
        h="56px"
        borderRadius="xl"
        bgColor={isActive ? 'background.300' : 'background.700'}
        transition=".2s ease-in"
        opacity="1"
        _hover={{
          bgColor: 'background.300',
        }}
        justifyContent="center"
        alignItems="center"
        mb="8px"
      >
        {!isActive ? icon : iconActive}
      </Flex>
      <Heading size="xs" variant="quaternary">
        {title}
      </Heading>
    </Radio>
  );
};

export default memo(ItemCategory, isEqual);
