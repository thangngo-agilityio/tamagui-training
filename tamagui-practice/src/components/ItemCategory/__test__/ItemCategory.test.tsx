import { render } from '@testing-library/react';
import ItemCategory from '..';

describe('ItemCategory component', () => {
  it('Should match snapshot for ItemCategory', () => {
    const { container } = render(<ItemCategory title="test" />);

    expect(container).toMatchSnapshot();
  });
});
