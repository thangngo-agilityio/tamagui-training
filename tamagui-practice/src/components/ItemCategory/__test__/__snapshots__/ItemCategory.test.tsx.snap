// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCategory component Should match snapshot for ItemCategory 1`] = `
<div>
  <label
    class="chakra-radio css-c6m1hx"
  >
    <input
      class="chakra-radio__input"
      id="radio-:r0:"
      style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
      type="radio"
      value=""
    />
    <span
      aria-hidden="true"
      class="chakra-radio__control css-ssalds"
    />
    <span
      class="chakra-radio__label css-1y8kf23"
    >
      <div
        class="css-1f5g8ea"
      />
      <h2
        class="chakra-heading css-0"
      >
        test
      </h2>
    </span>
  </label>
</div>
`;
